DuckDuckGo/EmailWaitlistViewController.swift

```diff
@@ -0,0 +1,388 @@
+        waitlistActionButton.isHidden = false
+        waitlistActionButton.setTitle(UserText.emailWaitlistJoinWaitlist, for: .normal)
+
+        existingInviteCodeButton.isHidden = false
+        existingInviteCodeButton.setTitle(UserText.emailWaitlistHaveInviteCode, for: .normal)
+    }
+
+    private func renderJoinedQueueState() {
+        headerImageView.image = Constants.weHatchedImage
+
+        headerTitleLabel.text = UserText.emailWaitlistJoinedWaitlist
+
+        if EmailWaitlist.shared.showWaitlistNotification {
+            headerDescriptionTextView.attributedText = createAttributedWaitlistJoinedWithNotificationSummary()
+        } else {
+            headerDescriptionTextView.attributedText = createAttributedWaitlistJoinedWithoutNotificationSummary()
+        }
+
+        footerTextView.attributedText = createAttributedPrivacyGuaranteeString()
+
+        waitlistActionButton.isHidden = true
+        existingInviteCodeButton.isHidden = false
+    }
+
+    private func renderInBetaState() {
+        headerImageView.image = Constants.contactUsImage
+
+        headerTitleLabel.text = UserText.emailWaitlistInvited
+        headerDescriptionTextView.attributedText = createAttributedWaitlistInvitedSummary()
+        footerTextView.attributedText = createAttributedPrivacyGuaranteeString()
+
+        waitlistActionButton.isEnabled = true
+        waitlistActionButton.isHidden = false
+        waitlistActionButton.setTitle(UserText.emailWaitlistGetStarted, for: .normal)
+
+        existingInviteCodeButton.isHidden = true
+    }
+
+    private func joinWaitlist() {
+        waitlistActionButton.isEnabled = false
+
+        emailManager.joinWaitlist { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case .success:
+                self.renderCurrentWaitlistState()
+                self.promptForNotificationPermissions()
+            case .failure:
+                self.waitlistActionButton.isEnabled = true
+                ActionMessageView.present(message: UserText.emailWaitlistErrorJoining)
+            }
+        }
+    }
+
+    private func promptForNotificationPermissions() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            if settings.authorizationStatus == .denied {
+                self.renderCurrentWaitlistState()
+                return
+            }
+
+            DispatchQueue.main.async {
+                let alertController = UIAlertController(title: UserText.emailWaitlistNotificationPermissionTitle,
+                                                        message: UserText.emailWaitlistNotificationPermissionBody,
+                                                        preferredStyle: .alert)
+
+                alertController.addAction(title: UserText.emailWaitlistNotificationPermissionNoThanks, style: .cancel) {
+                    EmailWaitlist.shared.showWaitlistNotification = false
```

Comment:
This is `false` by default, but for debugging purposes I'm setting it explicitly here - it's surfaced in the debug menu to make it clear whether a decision has been made. Will likely end up removing this before merging.